### ISSUE TITLE
[feature request] random permutation

### ISSUE THREAD

**Issue by samuela** [google/jax#1526]
It would be nice if we could get `jax.random` functions analogous to `np.random.permutation` and `np.random.shuffle`.

**Comment by mattjj** [comment]
There is a [`random.shuffle`](https://github.com/google/jax/blob/03cd6564e944cf60e58172abe0668762fa05d7e4/jax/random.py#L357). A `random.permutation` could be a simple wrapper around that IIUC.

**Comment by MichaelMarien** [comment]
I'd be happy to take a look at this.
